libs/wire-api/src/Wire/API/Team/Member.hs

```diff
@@ -292,17 +380,30 @@ modelTeamMemberDelete = Doc.defineModel "teamDeleteData" $ do
   Doc.property "password" Doc.string' $
     Doc.description "The account password to authorise the deletion."
 
-instance FromJSON TeamMemberDeleteData where
-  parseJSON = withObject "team-member-delete-data" $ \o ->
-    TeamMemberDeleteData <$> (o .:? "password")
+makeLenses ''TeamMember'
+makeLenses ''TeamMemberList'
+makeLenses ''NewTeamMember'
+makeLenses ''TeamMemberDeleteData
 
-instance ToJSON TeamMemberDeleteData where
-  toJSON tmd =
-    object
-      [ "password" .= _tmdAuthPassword tmd
-      ]
+userId :: Lens' TeamMember UserId
+userId = newTeamMember . nUserId
 
-makeLenses ''TeamMember
-makeLenses ''TeamMemberList
-makeLenses ''NewTeamMember
-makeLenses ''TeamMemberDeleteData
+permissions :: Lens (TeamMember' tag1) (TeamMember' tag2) (PermissionType tag1) (PermissionType tag2)
+permissions = newTeamMember . nPermissions
+
+invitation :: Lens' TeamMember (Maybe (UserId, UTCTimeMillis))
+invitation = newTeamMember . nInvitation
+
+-- JSON serialisation utilities (TODO: remove after servantification)
```

Comment:
Since these are staying for the time being, we should turn the `TODO` into a `FUTUREWORK`.